lib/plist/parser.rb

```diff
@@ -91,7 +89,25 @@ def parse
         if @scanner.scan(COMMENT_START)
           @scanner.scan(COMMENT_END)
         elsif @scanner.scan(XMLDECL_PATTERN)
+          next unless defined?(Encoding)
+          next unless String.method_defined?(:force_encoding)
+
+          xml_encoding = @scanner[1].match(/(?:\A|\s)encoding=(?:"(.*?)"|'(.*?)')(?:\s|\Z)/)
+
+          next if xml_encoding.nil?
+
+          encoding = begin
+            Encoding.find(xml_encoding[1])
+          rescue ArgumentError
+            nil
+          end
+
+          next if encoding.nil?
+
+          # use the specified encoding for the rest of the file
+          @scanner.string = @scanner.rest.force_encoding(encoding)
```

Comment:
I like this solution! My only nitpick is that this doubles the size of this scanning code. What do you think about factoring out the code you added into a helper method so it is easier to follow? E.g.

```ruby
elsif @scanner.scan(XMLDECL_PATTERN)
  force_encoding_based_on_xml_decl(@scanner[1])
elsif @scanner.scan(DOCTYPE_PATTERN)
```